src/Tools/ExternalAccess/VisualDiagnostics/Contracts/IHotReloadDiagnosticSource.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VisualDiagnostics.Contracts;
+
+/// <summary>
+/// Source of hot reload diagnostics.
+/// </summary>
+internal interface IHotReloadDiagnosticSource
+{
+    /// <summary>
+    /// Text document for which diagnostics are provided.
+    /// </summary>
+    TextDocument TextDocument { get; }
```

Comment:
this seems a bit odd, as this would be returning a snapshot.  The core api this is modeled after (IDiagnosticSource) only returns a DocId or ProjectId.  Can this just be a DocId intead?